### DISCUSSION TITLE
MetaMask not working in Firefox with CSP enabled

### DISCUSSION THREAD

**Discussion by oanamun** [family/connectkit#302] [Q&A] [answered]
Hello! I am having issues connecting the MetaMask wallet in Firefox due to an issue on their side where my content security policy setup is preventing MetaMask from injecting its scripts. Due to this the ethereum object is not present on window so Metamask is not detected by ConnectKit.

![Screenshot 2023-09-20 at 20 35 00](https://github.com/family/connectkit/assets/17378978/1f52d4ec-66ab-4d60-9d9f-fddc0d480fad)

The bug is not fixed even though the issue has been open for years, but there is a way around this that you can see [here] where you can load the content of the blocked script directly in your app (https://github.com/MetaMask/metamask-extension/issues/3133#issuecomment-1025641185). However when I try this I get the following, rather funny, state in ConnectKit because it detects an injected provider from what I can tell.

![Screenshot 2023-09-20 at 20 51 11](https://github.com/family/connectkit/assets/17378978/f5a96e21-0da4-49db-b2a6-aa6324fa14a5)

![Screenshot 2023-09-20 at 15 37 06](https://github.com/family/connectkit/assets/17378978/c7483329-c3b7-40f6-b054-288f712407da)

Do you know what I can do in this situation?

**Comment by lochie** [comment, marked as answer]
This is a very weird edge case.. these buttons are enabled/disabled based on [wagmi](https://github.com/wagmi-dev/wagmi)'s `ready` state on their connectors so this topic might be better suited in their discussions as their library is what handles this sort of issues.
